app/scripts/controllers/network/network-controller.js

```diff
@@ -224,42 +236,49 @@ export default class NetworkController extends EventEmitter {
     }
   }
 
-  setRpcTarget(rpcUrl, chainId, ticker = 'ETH', nickname = '', rpcPrefs) {
-    assert.ok(
-      isPrefixedFormattedHexString(chainId),
-      `Invalid chain ID "${chainId}": invalid hex string.`,
-    );
-    assert.ok(
-      isSafeChainId(parseInt(chainId, 16)),
-      `Invalid chain ID "${chainId}": numerical value greater than max safe value.`,
```

Comment:
Thoughts on preserving this validation in the `upsertNetworkConfiguration` function?  I see that it's still being validated in the UI, but having it here could prevent state corruption if we call this function from anywhere else in the future and forget to add validation.

Reply:
Yep makes sense to me!

Reply:
add here: https://github.com/MetaMask/metamask-extension/pull/17421/commits/6dbe2ac83e0e2890d7670339979f8bc4e005e4e4. 

- [ ] TODO: to add/adjust tests

Reply:
fixed tests [here](https://github.com/MetaMask/metamask-extension/pull/17421/commits/74eb7f1130392665017269629d927f7b5ead5b1a)